### PR TITLE
fix(route): Aqara News

### LINKED ISSUE #12904: aqara 新闻出错
### 路由地址

```routes
/aqara/news
```

### 完整路由地址

```fullroutes
/aqara/news
```

### 相关文档

https://docs.rsshub.app/other.html#aqara

### 预期是什么？

显示新闻

### 实际发生了什么？

出错

### 部署

RSSHub 演示 (https://rsshub.app)

### 部署相关信息

_No response_

### 额外信息

```shell
Route requested: /news
Error message: Cannot read properties of null (reading 'slice')
Helpful Information to provide when opening issue:
Path: /news
Node version: v18.17.0
Git Hash: 6f13fc4
```

### 这不是重复的 issue

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues)，以确保该错误尚未被报告。

### PR DESCRIPTION
**Pull request by nczitzk**
## 该 PR 相关 Issue / Involved Issue

Close #12904

## 路由地址示例 / Example for the Proposed Route(s)

```routes
/aqara/cn/news
/aqara/en/news
/aqara/en/blog
```

## 新 RSS 路由检查表 / New RSS Route Checklist
  
- [x] 新的路由 New Route
  - [x] 跟随 [v2 路由规范](https://docs.rsshub.app/joinus/script-standard.html) Follows [v2 Script Standard](https://docs.rsshub.app/en/joinus/script-standard.html)
- [x] 文档说明 Documentation
  - [x] 中文文档 CN
  - [ ] 英文文档 EN
- [x] 全文获取 fulltext
  - [x] 使用缓存 Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [x] [日期和时间](https://docs.rsshub.app/joinus/pub-date.html) [date and time](https://docs.rsshub.app/en/joinus/pub-date.html)
  - [x] 可以解析 Parsed
  - [x] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added
- [ ] `Puppeteer`

### CODE REVIEW

File: lib/v2/aqara/news.js [hunk 1]
```diff
@@ -3,51 +3,51 @@ const cheerio = require('cheerio');
...
-        url: currentUrl,
-    });
+    const { data: response } = await got(currentUrl);
 
-    let items = response.data
-        .match(/parm.contentHerf = '(\d+)'/g)
+    const $ = cheerio.load(response);
+
+    let items = response
+        .match(/(parm\.newsTitle[\s\S]*?arr\.push\(parm\))/g)
```
Comment (TonyRL) [inline lib/v2/aqara/news.js:16]: Nothing can be matched

Comment (nczitzk) [reply lib/v2/aqara/news.js:16]: Testing locally works fine.

aqara

Comment (TonyRL) [reply lib/v2/aqara/news.js:16]: Visiting https://www.aqara.cn/news using IP from CN will be redirected to `https://www.aqara.cn/news` while the rest will be redirected to `https://www.aqara.com/en/about-us/news/`

Comment (nczitzk) [reply lib/v2/aqara/news.js:16]: WIP

File: lib/v2/aqara/post.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        const filterKeyword = decodeURI(filterMatches[3].split('/').pop());
+        const filterApiUrl = new URL(`${filterRegion}/${apiSlug}/${filterType}?search=${filterKeyword}`, rootUrl).href;
+
+        const { data: filterResponse } = await got(filterApiUrl);
+
+        const filter = filterResponse.pop();
+
+        if (filter?.id ?? undefined) {
+            filterName = filter?.name ?? filterKeyword;
+            currentUrl = filter?.link ?? currentUrl;
```
Comment (TonyRL) [inline lib/v2/aqara/post.js:32]: https://deepscan.io/dashboard/#view=project&tid=6244&pid=8135&bid=92448&subview=pull-request&prid=1697381

File: lib/v2/aqara/news.js [hunk 3]
```diff
@@ -3,51 +3,51 @@ const cheerio = require('cheerio');
 const { parseDate } = require('@/utils/parse-date');
 
 module.exports = async (ctx) => {
-    const limit = ctx.query.limit ? parseInt(ctx.query.limit) : 35;
+    const limit = ctx.query.limit ? parseInt(ctx.query.limit, 10) : 35;
 
     const rootUrl = 'https://www.aqara.com';
-    const currentUrl = `${rootUrl}/news`;
+    const currentUrl = new URL('news', rootUrl).href;
```
Comment (TonyRL) [inline lib/v2/aqara/news.js:9]: https://github.com/DIYgod/RSSHub/pull/12921/files#r1287307532
<b>.cn</b>

Comment (nczitzk) [reply lib/v2/aqara/news.js:9]: 👌

**Pull request merged**